components/CustomButton.tsx

```diff
@@ -0,0 +1,42 @@
+import React from "react";
+import { Button } from "./ui/button";
+import { LucideProps } from "lucide-react";
+
+interface ButtonProps {
+  variant?:
```

Comment:
These are all optional, which is not ideal. Please require the props that we need.

Reply:
How would we handle the different variants? 
a. button with text
b. button with icon
c. button with icon and text